src/mappings/DAOFactory/mapping.ts

```diff
@@ -0,0 +1,138 @@
+import { Address, BigInt } from '@graphprotocol/graph-ts';
+import {
+  setContractsInfo,
+  setTemplatesInfo,
+} from '../../contractsInfo';
+
+import { Avatar } from '../../types/Controller/Avatar';
+import { DAOFactory, InitialSchemesSet, NewOrg, ProxyCreated, SchemeInstance } from '../../types/DAOFactory/DAOFactory';
+import {
+  AvatarContract, ContractInfo, DAOFactoryContract,
+} from '../../types/schema';
+import { createTemplate, debug, fetchTemplateName, setContractInfo } from '../../utils';
+
+export function getDAOFactoryContract(address: Address): DAOFactoryContract {
+  let daoFactory = DAOFactoryContract.load(address.toHex()) as DAOFactoryContract;
+  if (daoFactory == null) {
+    daoFactory = new DAOFactoryContract(address.toHex());
+    daoFactory.address = address;
+    let daoFactoryContract = DAOFactory.bind(address);
+    daoFactory.packageName = daoFactoryContract.PACKAGE_NAME();
+    daoFactory.app = daoFactoryContract.app();
+    daoFactory.save();
+    setContractsInfo();
+    setTemplatesInfo();
+  }
+  return daoFactory;
+}
+
+export function handleNewOrg(event: NewOrg): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOFactoryContract(event.address);
+
+  let avatar = event.params._avatar;
+
+  // If the avatar already exists, early out
+  if (AvatarContract.load(avatar.toHex()) != null) {
+    return;
+  }
+
+  let avatarContract = Avatar.bind(avatar);
+  let controller = avatarContract.owner();
+  let reputation = avatarContract.nativeReputation();
+  let daoToken = avatarContract.nativeToken();
+  let arcVersion: string;
+
+  let avatarInfo = ContractInfo.load(avatar.toHex());
+  if (avatarInfo != null) {
+    avatarInfo.name = 'Avatar';
+    avatarInfo.alias = 'Avatar';
+    avatarInfo.save();
+    arcVersion = avatarInfo.version;
+  } else {
+    // If contractInfo doesn't exist something went wrong and should exit early
```

Comment:
is that won't be the case on any new org created via dao factory ?

Reply:
Not sure I understand the question, but the `ProxyCreated` event is fired before the NewOrg and thus make sure the contractInfo exists beforehand. This check is just to make sure the subgraph won't crash if somehow the contractInfo doesn't exist (though it always should).

Reply:
contractInfo is a preCompiled data. 
it does not hold avatar addresses.

Reply:
is there a test for this mapping ? and specific for this functionality ?

Reply:
It can be precompiled, but not necessarily only so, and it also had `Avatar` and all other contracts. There's no reason not to use it here now that we only auto index daos.

There is a short test for that. https://github.com/daostack/subgraph/blob/subgraph-experimental/test/0.1.1-rc.1/contractsInfo.spec.ts

We could add more though doesn't seem necessary as this is a very simple mapping.